postgres/datadog_checks/postgres/metrics_cache.py

```diff
@@ -72,12 +72,11 @@ def get_instance_metrics(self, version):
             if not self.config.dbm_enabled:
                 c_metrics = dict(c_metrics, **DBM_MIGRATED_METRICS)
             # select the right set of metrics to collect depending on postgres version
+            self.instance_metrics = dict(c_metrics)
+            if version >= V9_2:
+                self.instance_metrics = dict(self.instance_metrics, **NEWER_92_METRICS)
             if version >= V14:
-                self.instance_metrics = dict(c_metrics, **NEWER_92_METRICS, **NEWER_14_METRICS)
-            elif version >= V9_2:
-                self.instance_metrics = dict(c_metrics, **NEWER_92_METRICS)
-            else:
-                self.instance_metrics = dict(c_metrics)
+                self.instance_metrics = dict(self.instance_metrics, **NEWER_14_METRICS)
```

Comment:
This LGTM, but wondering do we have test coverage that confirms metrics for v14 are emitted correctly?

Reply:
I've added tests on the session count, idle time and kill count.
The other metrics are harder to test, abandoned would require to trigger a RST on the socket and fatal seems even harder to trigger https://github.com/postgres/postgres/blob/master/src/backend/utils/error/elog.c#L584